src/array_api_stubs/_draft/array_object.py

```diff
@@ -28,9 +31,10 @@ def dtype(self: array) -> Dtype:
         out: dtype
             array data type.
         """
+        ...
 
     @property
-    def device(self: array) -> Device:
+    def device(self) -> "Device":  # type: ignore[type-var]
```

Comment:
```suggestion
    def device(self) -> Any:
```

Type variables (such as `Device` here) don't really make sense when it's only used as return type without using them in either a generic class or another parameter within the function; this is the source of mypy's error here and why you need the `type: ignore`; there must _always_ be at least two of them.

Instead, unless you're willing to make the protocol itself generic w.r.t. the device (which might be a bit excessive here), then it's customary to use `Any`. I'm not sure if this `Device` type is used anywhere else, but if not you could even consider just repurposing it as an `Any` alias in order to keep its more descriptive name.

Reply:
> unless you're willing to make the protocol itself generic w.r.t. the device (which might be a bit excessive here),

This is planned, but for a followup PR. (See opening comment to this PR).

>  then it's customary to use Any

Alternatively, this PR introduces a Device Protocol which can be used instead of the `TypeVar`. No binding issues.

Reply:
> Alternatively, this PR introduces a Device Protocol which can be used instead of the TypeVar. No binding issues because it's a Protocol. When Array is made generic wrt Device we can change this back to a TypeVar.

I guess that would be sort of possible? Rereading the [Device Support](https://data-apis.org/array-api/latest/design_topics/device_support.html#syntax-for-device-assignment) docs concrete API implementations have _a lot_ of freedom regarding the implementation of their Device type, to the point where only `__eq__` (and `__ne__`) really matter? Honestly, with an interface that minimal it might just be worthwhile to stick to plain `object` rather than `Any` or a protocol, as there's very little (or nothing) to gain by usage of the latter two.

Reply:
Yes, currently a `Device` Protocol has only ``__eq__`` in it and as such is effectively equivalent to `object`.
But it's designing with the future in mind ("The choice to not include a standardized Device object may be revisited in a future revision of this standard.") and we'd need to alias `object` anyway to get the correct rendering on the docs...